src/datastore/postgres/PgDataStore.ts

```diff
@@ -728,7 +728,7 @@ export class PgDataStore implements DataStore, ProvisioningStore {
         let currentVersion = await this.getSchemaVersion();
         while (currentVersion < PgDataStore.LATEST_SCHEMA) {
             log.info(`Updating schema to v${currentVersion + 1}`);
-            // eslint-disable-next-line @typescript-eslint/no-var-requires
+            // eslint-disable-next-line @typescript-eslint/no-var-requires,@typescript-eslint/no-require-imports
             const runSchema = require(`./schema/v${currentVersion + 1}`).runSchema;
```

Comment:
For the sake of avoiding `require()`, consider using a dynamic `import` instead:
```suggestion
            const runSchema = (await import(`./schema/v${currentVersion + 1}`)).runSchema;
```

Reply:
As per https://github.com/matrix-org/matrix-appservice-irc/pull/1858#discussion_r2459426450 I would like to avoid doing more adjustments to things, if possible, given time sensitivity and getting this bugfix done has already taken more time than was allocated to it. The change could be done as a separate pr to clean up, as it's not a functional change and thus doesn't need to go in to the next release.